### PR TITLE
Add port to database and UI.

### LINKED ISSUE #403: URL does not show port number
## :writing_hand: Describe the bug
If a URL includes a port number, it should be shown as part of the url, just like charles proxy

### PR DESCRIPTION
**Pull request by JayNewstrom**
Fixes #403

## :camera: Screenshots
Screenshot_20200803-215033

## :page_facing_up: Context
#403

## :no_entry_sign: Breaking
This will cause the database to be dropped/recreated on the next run.

## :hammer_and_wrench: How to test
I updated the automated tests, as well as tested manually via the UI (screenshot attached)

## :stopwatch: Next steps
Maybe add something to release notes about the database being dropped.

### CODE REVIEW

**Comment by vbuberen** [comment]
I guess this is not what was requested in #403 
The idea was to show port as part of URL, like Charles Proxy does. 

So the idea is to add port in to `URL` field on `Overview` and on `Network` screen, since it is really a part of URL

**Comment by JayNewstrom** [comment]
I've update the port to be stored/shown with the URL. This makes it so that it behaves more inline with the other parts of the URL, and also makes it so there's no breaking database changes.

**Comment by vbuberen** [comment]
I have 2 concerns here:
1. We might not want to always show port. For example, Charles Proxy shows port numbers not always, but when they are really part of the url (at least this is what it looks like to me). With current implementation we will always show `443` or `80` for https/http requests respectively.
2. If we add port into url we should also show the same url on the main screen with list of transactions.

**Comment by JayNewstrom** [comment]
> If we add port into url we should also show the same url on the main screen with list of transactions.
If we want to add it to the transactions list, we'd have to read/parse the url in `HttpTransactionTuple`, which I'm guessing is not desired for scrollability/speed. If you decide it should go there as well after this new info, let me know and I'll update it.

**Comment by JayNewstrom** [comment]
The other (hacky) thing I could do is store the port with the `host`, in the displayable format, since it looks like that is only used for that one thing.

**Comment by vbuberen** [comment]
> If we want to add it to the transactions list, we'd have to read/parse the url in `HttpTransactionTuple`, which I'm guessing is not desired for scrollability/speed. If you decide it should go there as well after this new info, let me know and I'll update it.

No need to update, from my point of view. We are going to redesign our DB entities for 4.x release (which we plan after 3.3.0). And in that change of DB it highly likely that we will have separate entities for transaction itself, request and response (at least this is what I am still thinking about and we discuss in #259 )
Except that I am going to do some changes on that transactions list as well.

As to your change about omitting 443 and 80 - looks good. I am curious to get more opinions there, but for me current options looks good enough.

**Comment by JayNewstrom** [comment]
Sounds good. Let me know if there's anything else I can do.

File: library/src/main/java/com/chuckerteam/chucker/internal/support/ShareUtils.kt [hunk 1]
```diff
@@ -94,7 +94,7 @@ internal object ShareUtils {
             // try to keep to a single line and use a subshell to preserve any line breaks
             curlCmd += " --data $'${requestBody.replace("\n", "\\n")}'"
         }
-        curlCmd += (if (compressed) " --compressed " else " ") + transaction.url
+        curlCmd += (if (compressed) " --compressed " else " ") + transaction.getFormattedUrl(false)
```
Comment (MiSikora) [inline library/src/main/java/com/chuckerteam/chucker/internal/support/ShareUtils.kt:97]: I suggest using a named parameter here.

Comment (JayNewstrom) [reply library/src/main/java/com/chuckerteam/chucker/internal/support/ShareUtils.kt]: Sure, updated.

**Comment by ColtonIdle** [comment]
Looks great and yes, having it part of the URL matches what I've gotten used to with Charles, so I think that makes sense.

**Pull request merged**